### DISCUSSION TITLE
Two AES keys for the same meter (Techem Radio 4)

### DISCUSSION THREAD

**Discussion by Tom1000** [wmbusmeters/wmbusmeters#1312] [Q&A]
I am renting multiple water meters with [Radio 4 module](https://oms-group.org/fileadmin/products/techem/Funkmodul_FM_radio_4/84964-01-0222-Funkmodul-FM-radio-4-INT-EN-2.pdf) from TECHEM in Germany, and I received AES keys from them to decode the telegrams. Similar to [some earlier reporting](https://github.com/wmbusmeters/wmbusmeters/issues/672#issuecomment-1821426749) I also received **two AES keys per meter**.

The meter send two messages a few seconds apart, both encrypted with the different keys. However, **how can I create the correct meter files to distinguish the telegrams from each other**? I saw [this post](url) with the hint to distinguish by type/version, but they seem to be the same in my case:

![image](https://github.com/wmbusmeters/wmbusmeters/assets/13649176/9ae478ac-285a-4675-8de6-48de81223008)
_left: first shorter message     right: longer message received a few seconds later_

So both messages differ in length and RSSI, but all other parameters seem to be the same:
`(telegram) DLL L=36 C=44 (from meter SND_NR) M=5068 (TCH) A=23306862 VER=95 TYPE=37 (Radio converter (meter side)) (driver mkradio4a) DEV=amb8465[00000000] RSSI=-91`
and:
`(telegram) DLL L=56 C=44 (from meter SND_NR) M=5068 (TCH) A=23306862 VER=95 TYPE=37 (Radio converter (meter side)) (driver mkradio4a) DEV=amb8465[00000000] RSSI=-103`

My wmbusmeters version: 1.13.1-3

P.S.: driver mkradio4 and mkradio4a are unable to completely decode the longer telegram. The best option seems to be evo868 but still a few bytes are not being decoded ([sample message here](https://wmbusmeters.org/analyze/5644685062683023953772106679732423fe07e60040052f2f4315390000426cff2c8304150B000082046cff238d04951f1e72fe030004000400040005000600060006000300000001000600080006000413012200002f:evo868)).

**Comment by rkorn86** [comment]
Hey Tom,

Sorry, I cannot answer your questions, but 
would you mind sharing what you have written to whom to get those keys directly without hassle?

**Reply by rkorn86** [reply]
I have got the following reply from Techem after asking for the AES Key:

> wir können den AES-Key aktuell nicht freigeben, da dies nicht unserer technischen und regulatorischen Vorgehensweise entspricht. Die Schlüssel sind Teil der Sicherheitsarchitektur des Messsystems und dürfen nicht extern weitergegeben werden, um die Integrität und Manipulationssicherheit der Verbrauchsdaten zu gewährleisten.

> es handelt sich hierbei aber nicht um ein Smartmeter, sondern um einen Wärmezähler in unserem Eigentum, welcher als Abrechnungsgrundlage dient und dessen Verschlüsselung Grundlage der Sicherheitsarchitektur des Messsystems ist.
 
> Wir haben Ihnen dargelegt, warum wir Ihnen den AES-Key nicht zur Verfügung stellen. Auch haben wir Ihnen dargelegt, wie wir die gewünschten Verbrauchsdaten zukünftig bereitstellen wollen.

> Weitergehend werden wir uns zum Thema Überlassung AES-Key nicht äußern und auch keine weiteren Nachfragen hierzu beantworten.

**Reply by Tom1000** [reply]
In my case, I only rent the meters and not the full service ("Abrechnung") from the company, therefore it might have been different.

**Reply by lenucksi** [reply]
Since those devices very likely are linux based / foss based at some point (receiver/sender), it might be an interesting way of trying to leverage the FOSS compliance requirements for the meters to see what's in there. Likely it'd be required to find some way of ownership of one of these devices for it though.

Also, it'd be interesting to see if there are individual keys per meter / meter-receiver combination or if they chose one for everything. (Get two people with meters, have them get the keys, hash the keys, compare the hashes)

**Reply by gerhardvilsecker** [reply]
Hello Tom, 
I just installed a brandnew Techem radio4 watermeter. I bought the device (product number: 10143510) from an ebay seller. 
Unfortunately this device seems not to send frames at all. My wmbusmeters environment gets a lot om OMS frames - but not from this device. May I ask if  there is anything to do in order to bring this device to "talking" mode? How frequent does your meter send data? Any help would be very appreciated. OMS keys should be no problem because I am owner of the device.
 
Thanks

Gerhard
